net_setup.go

```diff
@@ -0,0 +1,118 @@
+		GatewayIPv4 string
+		GatewayIPv6 string
+		GatewayIPv4WithCIDR string
+		GatewayIPv6WithCIDR string
+		DnsmasqEnabled bool
+	}
+	maskLenIPv4, _ := wireguardConfig.networkIPv4.Mask.Size()
+	maskLenIPv6, _ := wireguardConfig.networkIPv6.Mask.Size()
+	v := value{
+		Nameserver:          nameserver,
+		NetworkIPv4:         networkIPv4,
+		IPv6NatEnabled:      ipv6NatEnabled,
+		NetworkIPv6:         networkIPv6,
+		GatewayIPv4:         wireguardConfig.gatewayIPv4.String(),
+		GatewayIPv6:         wireguardConfig.gatewayIPv6.String(),
+		GatewayIPv4WithCIDR: fmt.Sprintf("%s/%d", wireguardConfig.gatewayIPv4.String(), maskLenIPv4),
+		GatewayIPv6WithCIDR: fmt.Sprintf("%s/%d", wireguardConfig.gatewayIPv6.String(), maskLenIPv6),
+		DnsmasqEnabled: enableDnsmasq,
+	}
+	return bash(`
+# Set DNS server
+echo "nameserver {{.Nameserver}}" > /etc/resolv.conf
+
+# Setup NAT
+
+## IPv4
+if ! /sbin/iptables -t nat --check POSTROUTING -s "{{.NetworkIPv4}}" -j MASQUERADE; then
+  /sbin/iptables -t nat --append POSTROUTING -s "{{.NetworkIPv4}}" -j MASQUERADE
+fi
+if ! /sbin/iptables --check FORWARD -m state --state RELATED,ESTABLISHED -j ACCEPT; then
+  /sbin/iptables --append FORWARD -m state --state RELATED,ESTABLISHED -j ACCEPT
+fi
+if ! /sbin/iptables --check FORWARD -s "{{.NetworkIPv4}}" -j ACCEPT; then
+  /sbin/iptables --append FORWARD -s "{{.NetworkIPv4}}" -j ACCEPT
+fi
+
+{{if .IPv6NatEnabled}}
+## IPv6
+if ! /sbin/ip6tables -t nat --check POSTROUTING -s "{{.NetworkIPv6}}" -j MASQUERADE; then
+  /sbin/ip6tables -t nat --append POSTROUTING -s "{{.NetworkIPv6}}" -j MASQUERADE
+fi
+if ! /sbin/ip6tables --check FORWARD -m state --state RELATED,ESTABLISHED -j ACCEPT; then
+  /sbin/ip6tables --append FORWARD -m state --state RELATED,ESTABLISHED -j ACCEPT
+fi
+if ! /sbin/ip6tables --check FORWARD -s "{{.NetworkIPv6}}" -j ACCEPT; then
+  /sbin/ip6tables --append FORWARD -s "{{.NetworkIPv6}}" -j ACCEPT
+fi
+{{end}}
+
+# ipv4 - DNS Leak Protection
+if ! /sbin/iptables -t nat --check OUTPUT -s "{{.NetworkIPv4}}" -p udp --dport 53 -j DNAT --to "{{.GatewayIPv4}}:53"; then
+  /sbin/iptables -t nat --append OUTPUT -s "{{.NetworkIPv4}}" -p udp --dport 53 -j DNAT --to "{{.GatewayIPv4}}:53"
+fi
+if ! /sbin/iptables -t nat --check OUTPUT -s "{{.NetworkIPv4}}" -p tcp --dport 53 -j DNAT --to "{{.GatewayIPv4}}:53"; then
+  /sbin/iptables -t nat --append OUTPUT -s "{{.NetworkIPv4}}" -p tcp --dport 53 -j DNAT --to "{{.GatewayIPv4}}:53"
+fi
+
+# ipv6 - DNS Leak Protection
+if ! /sbin/ip6tables --wait -t nat --check OUTPUT -s "{{.NetworkIPv6}}" -p udp --dport 53 -j DNAT --to "{{.GatewayIPv6}}"; then
+  /sbin/ip6tables --wait -t nat --append OUTPUT -s "{{.NetworkIPv6}}" -p udp --dport 53 -j DNAT --to "{{.GatewayIPv6}}"
+fi
+if ! /sbin/ip6tables --wait -t nat --check OUTPUT -s "{{.NetworkIPv6}}" -p tcp --dport 53 -j DNAT --to "{{.GatewayIPv6}}"; then
+  /sbin/ip6tables --wait -t nat --append OUTPUT -s "{{.NetworkIPv6}}" -p tcp --dport 53 -j DNAT --to "{{.GatewayIPv6}}"
+fi
```

Comment:
can we put this inside the IPV6NatEnabled piece, as not all boxes will have iptablesv6 installed